### PR TITLE
add test for remote signals after sbd server restart

### PR DESCRIPTION
**Pull request by neonphog**
RESOLVES: https://github.com/holochain/holochain/issues/4069

Wrote an integration test to reproduce #4069, but it works. My assumption is that this was fixed by https://github.com/holochain/holochain/pull/4133. But we still want the test for regressions.

### CODE REVIEW

File: crates/holochain/src/sweettest/sweet_conductor_config_rendezvous.rs [hunk 1]
```diff
@@ -77,28 +105,41 @@ impl SweetLocalRendezvous {
...
     }
+
+    /// Drop (shutdown) the signal server.
+    #[cfg(feature = "tx5")]
+    pub async fn drop_sig(&self) {
+        self.sig_hnd.lock().unwrap().take();
+
+        // NOTE: on windows (and slow other systems) we need to wait a moment
+        //       to make sure that the old connection is actually closed.
+        tokio::time::sleep(std::time::Duration::from_millis(400)).await;
```
Comment (jost-s) [inline crates/holochain/src/sweettest/sweet_conductor_config_rendezvous.rs:131]: Check here that the connection has been closed or put it in a retry loop?

File: crates/holochain/tests/tests/signals/mod.rs [hunk 2]
```diff
@@ -22,6 +22,113 @@ fn to_signal_message(signal: Signal) -> SignalMessage {
...
+    let (app1,) = c1
+        .setup_app_for_agent("app", a1.clone(), &[dna_file.clone()])
+        .await
+        .unwrap()
+        .into_tuple();
+    let (_app2,) = c2
+        .setup_app_for_agent("app", a2.clone(), &[dna_file])
+        .await
+        .unwrap()
+        .into_tuple();
```
Comment (jost-s) [inline crates/holochain/tests/tests/signals/mod.rs:57]: I think `SweetConductorBatch::from_rendezvous_config` with a `batch::setup_app` achieves that in fewer lines.

File: crates/holochain/tests/tests/signals/mod.rs [hunk 3]
```diff
@@ -22,6 +22,113 @@ fn to_signal_message(signal: Signal) -> SignalMessage {
...
+        .unwrap()
+        .value;
+
+    assert_eq!("hello", &msg);
+
+    // restart the signal (sbd) server so we get new ids
+    vous.start_sig().await;
+
+    // wait for that to propagate in holochain
+    tokio::time::sleep(std::time::Duration::from_secs(2)).await;
```
Comment (jost-s) [inline crates/holochain/tests/tests/signals/mod.rs:88]: Is there a way to check in Holochain that the signal server is online?

**Review by jost-s — CHANGES_REQUESTED** [review]

File: crates/holochain/tests/tests/signals/mod.rs (hunk 2 above)
Comment (neonphog) [reply crates/holochain/tests/tests/signals/mod.rs:57]: No, there's no way to specify the rendezvous instance to use with that method.

File: crates/holochain/tests/tests/signals/mod.rs (hunk 3 above)
Comment (neonphog) [reply crates/holochain/tests/tests/signals/mod.rs:88]: Nope.

File: crates/holochain/src/sweettest/sweet_conductor_config_rendezvous.rs (hunk 1 above)
Comment (neonphog) [reply crates/holochain/src/sweettest/sweet_conductor_config_rendezvous.rs]: done: https://github.com/holochain/holochain/pull/4151/commits/15274f5c53c0ca8350d1f4552954dfd377b9c967

**Comment by neonphog** [comment]
@jost-s - addressed comments

**Review by jost-s — APPROVED** [review]

**Pull request merged**